### PR TITLE
UI system/cpt 275

### PR DESCRIPTION
**Pull request by Chenziyes**
Add 4color player object material
Add screw driver and using logic with UI system
Update lock and unlock door

### CODE REVIEW

File: CSC8503/LevelManager.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ LevelManager::LevelManager() {
 
 	//preLoadtexID   I used Guard mesh to player and used rigMesh to guard   @(0v0)@  Chris 12/02/1998
 	mAnimation->PreloadMatTextures(*mRenderer, *mMeshes["Rig"], *mMaterials["Rig"], mGuardTextures);
-	mAnimation->PreloadMatTextures(*mRenderer, *mMeshes["Guard"], *mMaterials["Guard"], mPlayerTextures);
+	mAnimation->PreloadMatTextures(*mRenderer, *mMeshes["Guard"], *mMaterials["Guard_Red"], mPlayerTextures);
```
Comment (Nijelous) [inline CSC8503/LevelManager.cpp:71]: So I just changed how the whole material system worked with my most recent PR, so you may need to go through and change how this is loaded (now being done in the AddPlayerToWorld function)

File: CSC8503/LevelManager.h [hunk 2]
```diff
@@ -246,6 +246,10 @@ namespace NCL {
 #ifdef USEGL
 			vector<GLuint>  mGuardTextures;
 			vector<GLuint> mPlayerTextures;
+			vector<GLuint> mRedPlayerTextures;
+			vector<GLuint> mBluePlayerTextures;
+			vector<GLuint> mYellowPlayerTextures;
+			vector<GLuint> mGreenlayerTextures;
```
Comment (Nijelous) [inline CSC8503/LevelManager.h:252]: These are no longer necessary with my changes.

**Review by Nijelous — APPROVED** [review]
I mean the rest of it looks good just make sure you're using the new material system.

File: CSC8503/LevelManager.cpp (hunk 1 above)
Comment (Chenziyes) [reply CSC8503/LevelManager.cpp:71]: I noticed the change in the material module, it seems that just preloadMaterial is enough. 
Tomorrow I'll try to work with Eren on how to properly use the new material system and remove the redundancies!

File: CSC8503/LevelManager.h (hunk 2 above)
Comment (Chenziyes) [reply CSC8503/LevelManager.h:252]: Got it！

**Pull request merged**